doc/source/serve/autoscaling-guide.md

```diff
@@ -0,0 +1,369 @@
+
+However, the service latency rises to 400+ ms when the number of Locust users increased to 100.
+
+| P50 Latency | QPS |
+| ------- | --- |
+| ![comp_latency](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/model_comp_latency.svg) | ![comp_rps](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/model_comp_rps.svg) |
+
+Note that the number of Locust users should roughly match the number of `HeavyLoad` replicas to adequately serve the Locust traffic load. However, when the number of Locust users increased to 100, the `HeavyLoad` deployment struggled to get to 100 replicas, and instead only got to 65 replicas. A suspicion arises when we look at the per-deployment latencies. While `HeavyLoad` and `LightLoad` latencies stayed steady at 200ms and 100ms, `Driver` latencies rose from 220 to 400+ ms. This hints that hundreds of QPS may be harming the `Driver` replica's asynchronous event loop performance.
+
+
+### Attempt 2: Autoscale `Driver`
+
+Let's try setting an autoscaling configuration for `Driver` as well, with `target_num_ongoing_requests_per_replica = 20`. Now the deployment configurations are as follows.
+
+::::{tab-set}
+
+:::{tab-item} Driver
+
+```yaml
+- name: Driver
+  max_concurrent_queries: 200
+  autoscaling_config:
+    target_num_ongoing_requests_per_replica: 20
+    min_replicas: 1
+    initial_replicas: 1
+    max_replicas: 10
+    upscale_delay_s: 3
+    downscale_delay_s: 60
+    upscale_smoothing_factor: 0.3
+    downscale_smoothing_factor: 0.3
+    metrics_interval_s: 2
+    look_pack_period_s: 10
+```
+
+:::
+
+:::{tab-item} HeavyLoad
+
+```yaml
+- name: HeavyLoad
+  max_concurrent_queries: 3
+  autoscaling_config:
+    target_num_ongoing_requests_per_replica: 1
+    min_replicas: 0
+    initial_replicas: 0
+    max_replicas: 200
+    upscale_delay_s: 3
+    downscale_delay_s: 60
+    upscale_smoothing_factor: 0.3
+    downscale_smoothing_factor: 0.3
+    metrics_interval_s: 2
+    look_pack_period_s: 10
+```
+
+:::
+
+:::{tab-item} LightLoad
+
+```yaml
+- name: LightLoad
+  max_concurrent_queries: 3
+  autoscaling_config:
+    target_num_ongoing_requests_per_replica: 1
+    min_replicas: 0
+    initial_replicas: 0
+    max_replicas: 200
+    upscale_delay_s: 3
+    downscale_delay_s: 60
+    upscale_smoothing_factor: 0.3
+    downscale_smoothing_factor: 0.3
+    metrics_interval_s: 2
+    look_pack_period_s: 10
+```
+
+:::
+::::
+
+Running the same Locust load test again, we get the following results:
+
+| | |
+| ------------------------------------ | ------------------- |
+| HeavyLoad and LightLoad No. Replicas | <img src="https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/model_composition_improved_replicas.png" alt="heavy" width="500"/> |
```

Comment:
Can we switch `no.` to `number` throughout the tables? I think there's enough space to not need the abbreviation.